frontend/src/types/oauth.ts

```diff
@@ -45,7 +45,7 @@ export function openWindowForOAuth(
   return window.open(
     `${endpoint}?client_id=${applicationId}&redirect_uri=${encodeURIComponent(
       redirectUrl()
-    )}&state=${stateQueryParameter}&response_type=code&scope=api`,
+    )}&state=${stateQueryParameter}&response_type=code&scope=api,repo`,
```

Comment:
This is not optimal but works, both GitHub and GitLab ignores the scope they do not care.

To properly address this we need accept the VCS type for `openWindowForOAuth`. LMK if you want me to address it in this PR.

Reply:
We can do this is a separate PR